Interaccion pagina usuario

monkey-money-js opened the pull request.

monkey-money-js committed:
Agrega explicacion de tercer tarea de clase 5 r/ArgentinaPrograma

monkey-money-js committed:
Crea formulario para la tarea

monkey-money-js committed:
Modifica color y fuentes del formulario

monkey-money-js committed:
Centra y colorea primer input

monkey-money-js committed:
Crea el primer boton para saber de cuantos videos el usuario calculara el tiempo total

monkey-money-js committed:
Crea un node de inputs dependiendo de la cantidad de videos que quiera el usuario

monkey-money-js committed:
Cambia nombre de variable para evitar confusion  entre node y array

monkey-money-js committed:
Crea inputs de tiempo y los agrega a un div

Esto se hace para tener cada input por hora, minuto y segundo.

monkey-money-js committed:
Agrega la funcion de borrar divs de tiempo anteriores

Se hace para que no se acumulen divs en caso que el usuario toque el boton "siguiente" mas de una vez

monkey-money-js committed:
Agrega boton oculto para calcular el tiempo de los videos

monkey-money-js committed:
Borra parametro innecesario

monkey-money-js committed:
Muestra el boton de calcular tiempo cuando se hace click en 'Siguiente'

monkey-money-js committed:
Calcula el tiempo total de cada input que ingresa el usuario

monkey-money-js committed:
Agrega los resultados del tiempo total como texto a la pagina

monkey-money-js committed:
Arregla error en el que se puede obtener NaN como resultado en texto

monkey-money-js committed:
Revert "Arregla error en el que se puede obtener NaN como resultado en texto"

This reverts commit 61ed4b52491fbfd09047988c8d4a4564d59398d0.

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,111 @@
+const $botonSiguiente = document.querySelector('#boton-siguiente');
+const $botonCalcularTiempo = document.querySelector('#boton-calcular-tiempo');
+
+let $divVideos = document.querySelector('.div-videos');
+let $divResultados = document.querySelector('.resultados');
```
Estas dos variables pueden ser videos y la otra resultados, no es necesario aclarar el elemento. Si mañana deja de ser un div y es un p tenes que cambiar el elemento, la variable, y la clase.

De hecho, la clase de la linea 4 me parece que debería ser videos solo, como en la linea 5

Genebson replied:
Y debería ir con const esas 2 variables, pq en ningún momento cambian su valor, siempre es lo mismo.

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,111 @@
+
+$botonSiguiente.onclick = function(){
+    mostrarBotonTiempo();
+    let cantidadDeVideos = Number(document.querySelector('.cantidad-videos').value);
```
Acá igual, ``const``

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,111 @@
+}
+
+function borrarInputsAnteriores(){
+    const nodeDivDeTiempo = document.querySelectorAll('.divDeTiempo');
```
Acá lo mismo que mencioné sobre los elementos, y la clase tiene que ir en kebab-case y la palabra div no es necesaria, ya está implícita cuando creas el elemento

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,111 @@
+}
+
+function agregarInputsAlDiv(cantidadDeVideos){
+    let ternaDeInputs;
```
Qué significa ternaDeInputs? Puede ser const también no?

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,111 @@
+
+
+$botonCalcularTiempo.onclick = function(){
+    const arrayHoras = document.querySelectorAll('.horas');
```
Acá no hace falta aclarar que es un array, primero porque no lo es, es una lista de nodos, segundo si vos haces querySelectorAll('.loQueSea') estás haciendo referencia a que vas a agarrar MUCHOS elementos, por lo tanto sabes que va a ser una lista.

Entonces podes hacer const horas = document...
Lo mismo con las demás variables

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,111 @@
+    }
+}
+
+function crearDivInputsTiempo(){
```
Acá lo mismo, no hace falta aclarar el elemento. Lo que podes hacer es crear una función que se llame crearClases o algo así que haga referencia a las clases de los videos

Yo no recomendaría separar estas funciones (crearDivInputsTiempo, agregarInputsAlDiv) porque un div/label/input por si solas no tienen sentido (tienen baja coheción).

monkey-money-js committed:
Refactor de nombres de clases, declaraciones y nombres de variables

Se tomo en cuenta el feedback de un mentor de r/ArgentinaPrograma

monkey-money-js committed:
Refactor de dos funciones en una sola para crear inputs y agregarlos al div de una sola vez

Se hace para no separar la creacion de elementos y el uso de 'appendChild()' en dos funciones distintas, ya que crear un elemento por si solo no tiene sentido

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,106 @@
+    $botonCalcularTiempo.className = '';
+}
+
+function borrarInputsAnteriores(){
```
Me parece que ``borrarInputs`` como nombre de la función es suficiente

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,106 @@
+}
+
+function borrarInputsAnteriores(){
+    const nodeDeTiempo = document.querySelectorAll('.node-tiempo');
```
``const tiempo``

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,106 @@
+
+function crearClases(){
+
+    const $conjuntoTresInputs = document.createElement('div');
```
Que pasa si mañana dejan de ser 3 inputs y son 100?

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,106 @@
+    const inputSegundos = document.createElement('input');
+    inputSegundos.className = 'segundos';
+
+    const nodeDeInputs = [inputHoras, inputMinutos, inputSegundos];
```
``const inputs``

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,106 @@
+
+    const nodeDeInputs = [inputHoras, inputMinutos, inputSegundos];
+    
+    for (let i=0; i<3; i++){
```
Ese 3 debería ir en una constante. Qué significa ese 3? Qué representa?

monkey-money-js committed:
Refactor de nombres de variables y funciones

Genebson commented on tarea-3-de-clase-5.html:
```diff
@@ -0,0 +1,30 @@
+    <meta http-equiv='X-UA-Compatible' content='IE=edge'>
+    <title>Tarea 3 clase 5</title>
+    <meta name='viewport' content='width=device-width, initial-scale=1'>
+    <link rel='stylesheet' type='text/css' media='screen' href='tarea-3-de-clase-5.css'>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
Acá tenes que decidir si escribir todo con " o con '', es un detalle, pero es más que nada para mantener una consistencia en el código.

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,107 @@
+
+function crearClases(){
+
+    const $conjuntoInputs = document.createElement('div');
```
Si esto es un div que guarda todos los inputs dentro, se podría llamar `container`

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,107 @@
+    $conjuntoInputs.className = 'tiempo';
+    $conjuntoInputs.textContent = 'Ingrese horas, minutos y segundos';
+
+    const inputHoras = document.createElement('input');
```
Esta variable se podría llamar horas y lo mismo con las de abajo

Genebson commented on tarea-3-de-clase-5.js:
```diff
@@ -0,0 +1,107 @@
+    const inputSegundos = document.createElement('input');
+    inputSegundos.className = 'segundos';
+
+    const inputs = [inputHoras, inputMinutos, inputSegundos];
+    
+    const unidadesDeTiempo = 3;
+    for (let i=0; i<unidadesDeTiempo; i++){
+        $conjuntoInputs.appendChild(inputs[i]);
+    }
```
Qué sería unidadesDeTiempo? Los 3 inputs? Las horas/minutos/segundos? Si es eso podes hacer uso de la variable inputs y el método ``.length``
